components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/utils/SubscriptionValidationDataUtil.java

```diff
@@ -0,0 +1,248 @@
+
+                Set<String> groupIds = appModel.getGroupIds();
+                for (String grp : groupIds) {
+                    GroupIdDTO groupIdDTO = new GroupIdDTO();
+                    groupIdDTO.setApplicationId(appModel.getId());
+                    groupIdDTO.setGroupId(grp);
+                    applicationDTO.getGroupIds().add(groupIdDTO);
+                }
+
+                Map<String, String> attributes = appModel.getAttributesMap();
+                for (String attrib : attributes.keySet()) {
+                    ApplicationAttributeDTO applicationAttributeDTO = new ApplicationAttributeDTO();
+                    applicationAttributeDTO.setName(attrib);
+                    applicationAttributeDTO.setValue(attributes.get(attrib));
+                    applicationDTO.getAttributes().add(applicationAttributeDTO);
+                }
+                applicationListDTO.getList().add(applicationDTO);
+            }
+            applicationListDTO.setCount(model.size());
+
+        } else {
+            applicationListDTO.setCount(0);
+        }
+        return applicationListDTO;
+    }
+
+    public static SubscriptionListDTO fromSubscriptionToSubscriptionListDTO(List<Subscription> model) {
+
+        SubscriptionListDTO subscriptionListDTO = new SubscriptionListDTO();
+        if (model != null) {
+            for (Subscription subsModel : model) {
+                SubscriptionDTO subscriptionDTO = new SubscriptionDTO();
+                subscriptionDTO.setApiId(subsModel.getApiId());
+                subscriptionDTO.setAppId(subsModel.getAppId());
+                subscriptionDTO.setSubscriptionId(subsModel.getSubscriptionId());
+                subscriptionDTO.setPolicyId(subsModel.getPolicyId());
+                subscriptionDTO.setSubscriptionState(subsModel.getSubscriptionState());
+
+                subscriptionListDTO.getList().add(subscriptionDTO);
+
+            }
+            subscriptionListDTO.setCount(model.size());
+
+        } else {
+            subscriptionListDTO.setCount(0);
+        }
+        return subscriptionListDTO;
+    }
+
+    public static SubscriptionPolicyListDTO fromSubscriptionPolicyToSubscriptionPolicyListDTO(
+            List<SubscriptionPolicy> model) {
+
+        SubscriptionPolicyListDTO subscriptionPolicyListDTO = new SubscriptionPolicyListDTO();
+        if (model != null) {
+            for (SubscriptionPolicy subscriptionPolicyModel : model) {
+                SubscriptionPolicyDTO subscriptionPolicyDTO = new SubscriptionPolicyDTO();
+                subscriptionPolicyDTO.setId(subscriptionPolicyModel.getId());
+                subscriptionPolicyDTO.setName(subscriptionPolicyModel.getName());
+                subscriptionPolicyDTO.setQuotaType(subscriptionPolicyModel.getQuotaType());
+                subscriptionPolicyDTO.setTenantId(subscriptionPolicyModel.getTenantId());
+
+                subscriptionPolicyListDTO.getList().add(subscriptionPolicyDTO);
```

Comment:
rateLimitCount,rateLimitTimeUnit,stopOnQuotaReach are not set

Reply:
fixed